torch/_inductor/codegen/triton.py

```diff
@@ -2624,6 +2624,11 @@ def codegen_kernel(self, name=None):
                 mutated_args.add(self.args.inplace_buffers[mutation].inner_name)
             if mutation in self.args.output_buffers:
                 mutated_args.add(self.args.output_buffers[mutation])
+
+        for argname, arg in zip(argdefs, signature):
+            if isinstance(arg, WorkspaceArg) and arg.zero_fill:
```

Comment:
```suggestion
            if isinstance(arg, WorkspaceArg):
```

I don't think the zero-fill is relevant here.

Reply:
iiuc, if zero_fill is false, then you don't actually have any requirements on the initial values in the workspace, right? in which case, do we actually need to handle it as a mutable argument (when we don't actually care about its mutations)?

Reply:
Makes sense, worth a comment though as it's still mutating  we just don't need to restore it

Reply:
thanks - added a comment